lua/tests/current_search_paths_spec.lua

```diff
@@ -0,0 +1,49 @@
+local lean = require('lean')
+local helpers = require('tests.helpers')
+local fixtures = require('tests.fixtures')
+
+require('tests.helpers').setup {
+  lsp = { enable = true },
+  lsp3 = { enable = true },
+}
+describe('lean.current_search_paths', function()
+  for kind, path in pairs{
+    ["existing"] = fixtures.lean_project.some_existing_file,
+    ["nested existing"] = fixtures.lean_project.some_nested_existing_file,
+    ["nonexisting"] = fixtures.lean_project.some_nonexisting_file,
+    ["nested nonexisting"] = fixtures.lean_project.some_nested_nonexisting_file
+  } do
```

Comment:
Good idea putting these paths strings in a helper rather than duplicating. I think we can define a custom iterator for this in the helper as well (which we can also use in `ft_spec.lua`), will do.